app/controllers/audio_files_controller.rb

```diff
@@ -35,6 +39,17 @@ def destroy
   redirect_to audio_files_path
 end
 
+def update
+  @audio_file = AudioFile.find_by_id(params[:id])
+
+  if @audio_file.update(audio_file_params)
+    redirect_to @audio_file
+  else
+    render 'edit'
+    redirect_to @audio_file
```

Comment:
The `redirect_to` probably doesn't do what we want - we want to stay on this page and render the `edit` view, which line 48 is doing.